firestore-dialogflow-fulfillment/functions/src/index.ts

```diff
@@ -20,11 +20,86 @@ import { FieldValue } from "firebase-admin/firestore";
 import { WebhookClient } from "dialogflow-fulfillment-helper";
 import { HttpsError } from "firebase-functions/v1/auth";
 import DialogFlow from "@google-cloud/dialogflow";
+import { google } from "googleapis";
+import { GaxiosError } from "gaxios";
+
 import config from "./config";
+import Status from "./types/status";
+import Conversation from "./types/conversation";
+import { extratDate, getDateTimeFormatted } from "./util";
+
+var fs = require("fs");
 
-const dialogflow = DialogFlow.v2beta1;
 admin.initializeApp();
 
+const dialogflow = DialogFlow.v2beta1;
+const sessionClient = new dialogflow.SessionsClient({
+  projectId: config.projectId,
+  ...(fs.existsSync(config.servicePath) && { keyFilename: config.servicePath }),
```

Comment:
This needs to be removed for the extension - might have to provide an auth client instance.